src/pybind/mgr/prometheus/module.py

```diff
@@ -1151,16 +1151,13 @@ def get_file_sd_config(self):
                 if service['type'] != 'mgr':
                     continue
                 id_ = service['id']
-                # get port for prometheus module at mgr with id_
-                # TODO use get_config_prefix or get_config here once
-                # https://github.com/ceph/ceph/pull/20458 is merged
                 result = CommandResult("")
                 assert isinstance(_global_instance, Module)
                 _global_instance.send_command(
                     result, "mon", '',
                     json.dumps({
-                        "prefix": "config-key get",
-                        'key': "config/mgr/mgr/prometheus/{}/server_port".format(id_),
+                        "prefix": "config get",
+                        'key': "mgr.{} mgr/prometheus/server_port".format(id_),
```

Comment:
```suggestion
                        'who': f'mgr.{id_}',
                        'key': 'mgr/prometheus/server_port',
```

see https://github.com/ceph/ceph/blob/4040082610b25990e9e6ca0d0e891e8e0dce574c/src/mon/MonCommands.h#L1279-L1283

Reply:
also, the title could be more specific:

> pybind/mgr/prometheus: use "config get" for retrieving options

Reply:
@Rethan please update the title of the commit message, use `git commit --amend`  and `git push -f` for revising the commit message and its title.

Reply:
https://github.com/ceph/ceph/pull/38225/files#r528209717 is not addressed.

Reply:
It seems that test code(which is `bin/ceph mgr selftest module prometheus`) fails after introducing the second commit. And vstart.sh line 970 and mgr_test_case.py "set_module_localized_conf" still use the way of `config set mgr/prometheus/{mgr_id}/server_port` to set that key. Should we take a step back to minimize changes or a step further to change other things accordingly?

Reply:
ahh, right. so the way how we retrieve the setting is not backward compatible. i think we should offer an upgrade path to user who are using `config/mgr/mgr/prometheus/{id}/server_port`.

Reply:
So, for compatibility, maybe we can go back to the old-fasioned way.